File: java/Observer/empty/Observable.java
import java.util.ArrayList;

/*
the "subject" or "observable" class, state changes in this class are sent to
the observer
*/
public class Observable {

  //initialize list of observer and state

  //add an observer and give it Observable's current state
  public void add_observer(Observer obs){}

  //remove an observer from our list
  public void remove_observer(Observer obs){}

  //notify all observers about our current state
  public void notify_observers(){}

  //set the current state and notify our Observers of the change
  public void set_state(String state){}
}

File: python/Observer/empty/Observable.py
from Observer import Observer

class Observable(object):
    '''
    the 'subject' or 'observable' class, state changes in this class are sent to the
    observers
    '''

    def __init__(self):
        #initialize our list of observers and our state

    def add_observer(self, obs):
        #add an observer to our list of observers and notify it of current state

    def remove_observer(self, obs):
        #remove an observer from our list

    def notify_observers(self):
        #notify observers of current state

    def set_state(self, state):
        #set the current state and notify observers of change
